autopush_rs/src/protocol.rs

```diff
@@ -120,6 +123,33 @@ pub struct Notification {
     pub headers: Option<HashMap<String, String>>,
 }
 
+impl Notification {
+    /// Return an appropriate sort_key to use for the chidmessageid
+    ///
+    /// For new messages:
+    ///     02:{sortkey_timestamp}:{chid}
+    ///
+    /// For topic messages:
+    ///     01:{chid}:{topic}
+    ///
+    /// Old format for non-topic messages that is no longer returned:
+    ///     {chid}:{message_id}
+    pub fn sort_key(&self) -> String {
+        let chid = self.channel_id.hyphenated();
+        if let Some(ref topic) = self.topic {
+            format!("01:{}:{}", chid, topic)
+        } else if let Some(ref sortkey_timestamp) = self.sortkey_timestamp {
```

Comment:
you don't even need the ref or derefing, you'll get a copy

Reply:
I only put that there cause Rust complained:
```
139 |         if let Some(topic) = self.topic {
    |                     -----    ^^^^ cannot move out of borrowed content
    |                     |
    |                     hint: to prevent move, use `ref topic` or `ref mut topic`
```